tests/integration/suite/test_rbac.py

```diff
@@ -495,9 +495,19 @@ def is_installing():
         action_name="upgrade",
         answers={"asdf": "asdf"})
 
-    '''
-        TODO: write rollback test, currently blocked by issue #20204
-    '''
+    def _app_revisions_exist():
+        a = admin_pc.client.reload(app)
+        return len(a.revision().data) > 0
+
+    wait_for(_app_revisions_exist, fail_handler=lambda: 'no revisions exist')
+    app = admin_pc.client.reload(app)
```

Comment:
This is the admin, who can always see app revisions, this needs to be based off the user client

Reply:
Technically the rollback would have failed anyway because the user can't access that revision, I tested and confirmed that, but I updated it to check here too. I'm leaving the app_revision check as `admin_pc` because a failure there would be ambiguous between a permissions issue or an infrastructure issue.

Reply:
Technically the test would have failed anyway because the user can't see the rollback, I tested and confirmed that, but I updated this to check too. I'm leaving the `_app_revisions_exist` check as `admin_pc` because a failure there would be ambiguous, it could mean either a permissions issue or an infrastructure issue.